plugins/flashrom/fu-flashrom-device.c

```diff
@@ -48,11 +72,206 @@ fu_flashrom_device_set_quirk_kv (FuDevice *device,
 	return FALSE;
 }
 
+static int
+fu_flashrom_device_debug_cb (enum flashrom_log_level lvl, const char *fmt, va_list args)
+{
+#pragma clang diagnostic push
+#pragma clang diagnostic ignored "-Wformat-nonliteral"
+	g_autofree gchar *tmp = g_strdup_vprintf (fmt, args);
+#pragma clang diagnostic pop
+	g_autofree gchar *str = fu_common_strstrip (tmp);
+	if (g_strcmp0 (str, "OK.") == 0 || g_strcmp0 (str, ".") == 0)
+		return 0;
+	switch (lvl) {
+	case FLASHROM_MSG_ERROR:
+	case FLASHROM_MSG_WARN:
+		g_warning ("%s", str);
+		break;
+	case FLASHROM_MSG_INFO:
+		g_debug ("%s", str);
+		break;
+	case FLASHROM_MSG_DEBUG:
+	case FLASHROM_MSG_DEBUG2:
+		if (g_getenv ("FWUPD_FLASHROM_VERBOSE") != NULL)
+			g_debug ("%s", str);
+		break;
+	case FLASHROM_MSG_SPEW:
+		break;
+	default:
+		break;
+	}
+	return 0;
+}
+
+static gboolean
+fu_flashrom_device_setup (FuDevice *device, GError **error)
+{
+	FuFlashromDevicePrivate *priv = GET_PRIVATE (FU_FLASHROM_DEVICE (device));
+	gint rc;
+
+	if (flashrom_init (SELFCHECK_TRUE)) {
+		g_set_error_literal (error,
+				     FWUPD_ERROR,
+				     FWUPD_ERROR_NOT_SUPPORTED,
+				     "flashrom initialization error");
+		return FALSE;
+	}
+	flashrom_set_log_callback (fu_flashrom_device_debug_cb);
```

Comment:
isn't this global?

Reply:
@sammc @bleungatchromium do you have any insights on this?